src/spikeinterface/core/channelsaggregationrecording.py

```diff
@@ -13,7 +13,16 @@ class ChannelsAggregationRecording(BaseRecording):
 
     """
 
-    def __init__(self, recording_list, renamed_channel_ids=None):
+    def __init__(self, recording_list_or_dict, renamed_channel_ids=None):
+
+        if isinstance(recording_list_or_dict, dict):
+            recording_list = list(recording_list_or_dict.values())
```

Comment:
```suggestion
            recording_list = list(recording_list_or_dict.values())
```
In this case, I think it would make sense to check if each recording's group corresponds to the key.
If all groups are 0 (which is default), we could set channel groups according to keys

Reply:
At the moment there is nothing special about "group". As in, a user can do
```
rec.split_by("user_defined_label")
```
And we don't keep track of the label since `split_by` just returns a dictionary. So this would make "group" a special label. I think this is ok: we can recommend users use it when we document `split_by`?

Reply:
You're abslutely right. We could actually track this as an annotation though :)

Reply:
Oh nice! Lemme have a go...